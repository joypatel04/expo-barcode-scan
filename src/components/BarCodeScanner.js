'use strict';

import React, { PropTypes, Component } from 'react';
import { Animated, StyleSheet, Text, View, TouchableOpacity, ScrollView } from 'react-native';
import { Ionicons } from '@expo/vector-icons'
import { ActionCreators } from './../actions';
import Store from './../store/configureStore';
import Router from './../Router';
import Emoji from'react-native-emoji';
import { NavigationActions } from '@expo/ex-navigation';

export default class BarCodeScanner extends Component {
    
    constructor(props) {
        super(props);
        this.state = {
            
        }
    }

    _addProduct = () => {
        let navigatorUID = Store.getState().navigation.currentNavigatorUID;
        Store.dispatch(NavigationActions.push(navigatorUID, Router.getRoute('addProduct', { title: 'Add Product' })))
    }

    _scanProduct = () => {
        let navigatorUID = Store.getState().navigation.currentNavigatorUID;
        Store.dispatch(NavigationActions.push(navigatorUID, Router.getRoute('scanProduct', { title: 'Scan Product' })))
    }

    _allProduct = () => {
        let navigatorUID = Store.getState().navigation.currentNavigatorUID;
        Store.dispatch(NavigationActions.push(navigatorUID, Router.getRoute('allProducts', { title: 'All Products' })))
    }

    render() {
        return (
            <ScrollView contentContainerStyle={[styles.container, { width: this.props.width}]} scrollEnabled={false}>
                <View style={[styles.header, {flexDirection: 'row', alignItems: 'center'}]}>
                    <Text style={[styles.headerTitle, {marginRight: 10}]}> Camera / Scanner Example </Text>
                    <TouchableOpacity onPress={() => this.props.onToggleModal()}>
                        <Ionicons name={'ios-information-circle'} size={30} color={'rgba(0, 0, 0, 0.75)'} />
                    </TouchableOpacity>
                </View>
                <View style={[styles.header, styles.rowDireaction,{ flexGrow: 1 }]}>
                    <TouchableOpacity onPress={() => this._addProduct()} style={styles.space}>
                        <Text style={{fontSize: 50}}><Emoji name=":inbox_tray:"/></Text>
                        <Text style={[styles.headerTitle]}> Add Product </Text>
                    </TouchableOpacity>
                    <TouchableOpacity onPress={() => this._scanProduct()} style={styles.space}>
                        <Text style={{fontSize: 50}}><Emoji name=":iphone:"/></Text>
                        <Text style={[styles.headerTitle]}> Scan Product </Text>
                    </TouchableOpacity>
                    <TouchableOpacity onPress={() => this._allProduct()} style={styles.space}>
                        <Text style={{fontSize: 50}}><Emoji name=":card_file_box:"/></Text>
                        <Text style={[styles.headerTitle]}> All Products </Text>
                    </TouchableOpacity>
                </View>
            </ScrollView>
        );
    }

}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: 'transparent',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  header: {
    padding: 10
  },
  headerTitle: {
    fontFamily: 'HelveticaNeue',
    fontSize: 16,
    fontWeight: '700',
    color: '#455A64',
    letterSpacing: 0
  },
  space: {
    marginTop: 10,
    marginBottom: 10,
    height: 80,
    justifyContent: 'center',
    alignItems: 'center'
  },
});