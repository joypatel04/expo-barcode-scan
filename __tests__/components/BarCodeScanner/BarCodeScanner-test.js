import 'react-native';
import React from 'react';
import BarCodeScanner from './../../../src/components/BarCodeScanner';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('Render BarCodeScanner Component Correctly', () => {
  const tree = renderer.create(
    <BarCodeScanner width={100} onToggleModal={() => {}} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})