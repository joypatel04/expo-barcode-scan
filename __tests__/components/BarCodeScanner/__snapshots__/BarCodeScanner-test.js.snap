// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render BarCodeScanner Component Correctly 1`] = `
<ScrollView
  contentContainerStyle={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "justifyContent": "flex-start",
      },
      Object {
        "width": 100,
      },
    ]
  }
  scrollEnabled={false}
>
  <View
    style={
      Array [
        Object {
          "padding": 10,
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#455A64",
            "fontFamily": "HelveticaNeue",
            "fontSize": 16,
            "fontWeight": "700",
            "letterSpacing": 0,
          },
          Object {
            "marginRight": 10,
          },
        ]
      }
    >
       Camera / Scanner Example 
    </Text>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "padding": 10,
        },
        undefined,
        Object {
          "flexGrow": 1,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 80,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 10,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 50,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          ðŸ“¥
        </Text>
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#455A64",
              "fontFamily": "HelveticaNeue",
              "fontSize": 16,
              "fontWeight": "700",
              "letterSpacing": 0,
            },
          ]
        }
      >
         Add Product 
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 80,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 10,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 50,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          ðŸ“±
        </Text>
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#455A64",
              "fontFamily": "HelveticaNeue",
              "fontSize": 16,
              "fontWeight": "700",
              "letterSpacing": 0,
            },
          ]
        }
      >
         Scan Product 
      </Text>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "height": 80,
          "justifyContent": "center",
          "marginBottom": 10,
          "marginTop": 10,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 50,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          ðŸ—ƒ
        </Text>
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#455A64",
              "fontFamily": "HelveticaNeue",
              "fontSize": 16,
              "fontWeight": "700",
              "letterSpacing": 0,
            },
          ]
        }
      >
         All Products 
      </Text>
    </View>
  </View>
</ScrollView>
`;
